sklearn/ensemble/tests/test_gradient_boosting_loss_functions.py

```diff
@@ -145,6 +148,45 @@ def test_sample_weight_deviance():
         assert deviance_wo_w == deviance_w_w
 
 
+@pytest.mark.parametrize(
+    'n_classes, n_samples', [(3, 100), (5, 57), (7, 13)]
+)
+def test_multinomial_deviance(n_classes, n_samples):
+    # Check multinomial deviance with and without sample weights.
+    rng = np.random.RandomState(13)
+    sample_weight = np.ones(n_samples)
+    y_true = rng.randint(0, n_classes, size=n_samples)
+    y_pred = np.zeros((n_samples, n_classes), dtype=np.float64)
+    for klass in range(y_pred.shape[1]):
+        y_pred[:, klass] = y_true == klass
+
+    loss = MultinomialDeviance(n_classes)
+    loss_wo_sw = loss(y_true, y_pred)
+    assert loss_wo_sw > 0
+    loss_w_sw = loss(y_true, y_pred, sample_weight=sample_weight)
+    assert loss_wo_sw == pytest.approx(loss_w_sw)
+    loss_w_sw = loss(y_true, y_pred, sample_weight=0.5 * sample_weight)
+    assert loss_wo_sw == pytest.approx(loss_w_sw)
+
+
+def test_mdl_computation_weighted():
+    y_pred = np.array([[1.0, 0, 0], [0, 0.5, 0.5]])
```

Comment:
This should be the (unormalized) raw predictions, that is the logits, before going through the inverse link functio (the softmax in this case). They have no reasons to lie between 0 and 1.

So I would suggest to rewrite this test with values such as:

```
raw_predictions = np.array([[1., -1., -.1], [-2., 1., 2.]])
```

for instance.

Reply:
You're certainly right. I applied your suggestion.